services/merak-topo/handler/topo_deploy.go

```diff
@@ -28,72 +26,60 @@ import (
 	"github.com/futurewei-cloud/merak/services/merak-topo/utils"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/client-go/dynamic"
 	"k8s.io/client-go/kubernetes"
-
-	"k8s.io/client-go/tools/remotecommand"
 )
 
 var (
 	SDN_IP   = "sdn-controller.merak.svc.cluster.local"
 	SDN_PORT = "6653"
 	Ctx      = context.Background()
 
-	namespace        = "default"
 	topologyClassGVR = schema.GroupVersionResource{
 		Group:    "networkop.co.uk",
 		Version:  "v1beta1",
 		Resource: "topologies",
 	}
 )
 
-func CreateTopologyClasses(client dynamic.Interface, name string, links []database.Vlink) error {
-	rc := NewTopologyClass(name, links)
+func CreateTopologyClasses(client dynamic.Interface, name string, links []database.Vlink, namespace string) error {
+	rc := NewTopologyClass(name, links, namespace)
+
+	var err_return error
 
 	_, err := client.Resource(topologyClassGVR).Namespace(namespace).Create(Ctx, rc, metav1.CreateOptions{})
 
 	if err != nil {
-		return fmt.Errorf("failed to create topologyClass %s", err)
+		utils.Logger.Error("can't create topologyClass", "create topology class error", err.Error(), "namespace", namespace, "vnode", name)
+		err_return = err
```

Comment:
@zzengcs just `return err_return` should be enough.

Reply:
addressed